pkg/iplist/github/github_test.go

```diff
@@ -0,0 +1,36 @@
+package github
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWithURL(t *testing.T) {
+	t.Parallel()
+
+	testUrl := "test.example.com"
+
+	g := New(WithURL(testUrl))
+	assert.Equal(t, testUrl, g.url)
+}
+
+func TestWithTimeout(t *testing.T) {
+	t.Parallel()
+
+	testTimeout := time.Hour * 123
+
+	g := New(WithTimeout(testTimeout))
+	assert.Equal(t, testTimeout, g.timeout)
+}
+
+func TestGetIPs(t *testing.T) {
+	t.Parallel()
+
+	g := New()
+	res, err := g.GetIPs()
```

Comment:
I left this test as the actual getting IPs from github instead of mock because really that's the point of the function. If github changes their API, we should know this test fails.

The officeips_test.go is mocked because we don't want to expose the private endpoint in this public repo. But I have a similar feeling for that one, altho nothing can do.

Let me know if you feel different

Reply:
I think it should be mocked.  We do not necessarily want the tests to fail if github changes their IPs or API.  The controller running this also needs to surface the error and fail gracefully. 

Alternatively I would at least mock this as an integration test to make clear that it is making an external call.

Reply:
I ended up using mock responses throughout.

I can add integration tests in a subsequent PR as this one is getting big.